src/main/java/io/tiledb/spark/StringPartitioner.java

```diff
@@ -0,0 +1,147 @@
+package io.tiledb.spark;
+
+import io.tiledb.java.api.Pair;
+import java.util.ArrayList;
+import java.util.List;
+
+public class StringPartitioner {
+  private int MIN_ASCII_CHAR = 0;
+  private int MAX_ASCII_CHAR = 127;
+
+  private int minChar;
+  private int maxChar;
+
+  public StringPartitioner() {
+    minChar = MIN_ASCII_CHAR;
+    maxChar = MAX_ASCII_CHAR;
+  }
+
+  public StringPartitioner(int minChar, int maxChar) {
+    this.minChar = minChar;
+    this.maxChar = maxChar;
+  }
+
+  /**
+   * Computes the next possible string by incrementing the least possible character. The next string
+   * is limited to the input string length. For example, if the minimum character provided is 'z',
+   * then a string like `zzz` cannot be further incremented and it will be returned as it is. On the
+   * other hand, the next string of `aaz` will be `aba`, the next of `a` `b` and so on.
```

Comment:
Add a note that your examples here are limited to a-z for simplicity of the example but go from 0-127 ascii characters in actual code.